public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx

```diff
@@ -25,38 +21,52 @@ const forValidationOptions = (evaluateEvery: string): RegisterOptions => ({
     value: true,
     message: 'Required.',
   },
-  pattern: durationValidationPattern,
-  validate: (value) => {
-    const millisFor = parseDurationToMilliseconds(value);
-    const millisEvery = parseDurationToMilliseconds(evaluateEvery);
+  validate: (value: string) => {
+    try {
+      const millisFor = parsePrometheusDuration(value);
 
-    // 0 is a special value meaning for equals evaluation interval
-    if (millisFor === 0) {
-      return true;
-    }
+      // 0 is a special value meaning for equals evaluation interval
+      if (millisFor === 0) {
+        return true;
+      }
 
-    return millisFor >= millisEvery ? true : 'For must be greater than or equal to evaluate every.';
+      try {
+        const millisEvery = parsePrometheusDuration(evaluateEvery);
+        return millisFor >= millisEvery
+          ? true
+          : 'For duration must be greater than or equal to the evaluation interval.';
+      } catch (err) {
+        // if we fail to parse "every", assume validation is successful, or the error messages
+        // will overlap in the UI
+        return true;
+      }
+    } catch (error) {
+      return error instanceof Error ? error.message : 'Failed to parse duration';
+    }
   },
 });
 
-const evaluateEveryValidationOptions: RegisterOptions = {
+export const evaluateEveryValidationOptions: RegisterOptions = {
   required: {
     value: true,
     message: 'Required.',
   },
-  pattern: positiveDurationValidationPattern,
   validate: (value: string) => {
-    const duration = parseDuration(value);
-    if (Object.keys(duration).length) {
-      const diff = durationToMilliseconds(duration);
-      if (diff < MIN_TIME_RANGE_STEP_S * 1000) {
+    try {
+      const duration = parsePrometheusDuration(value);
```

Comment:
I would name this const  with`millisEvery` or `millisEvaluateEvery` to be consistent with the  `For`, and I think it would be more clear